test/integration/supervisor_tls_spec_test.go

```diff
@@ -0,0 +1,266 @@
+package integration
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.pinniped.dev/internal/here"
+)
+
+// TestTLSSpecKubeBuilderValidationSupervisor_Parallel tests kubebuilder validation
+// on the TLSSpec in Pinniped supervisor CRDs using OIDCIdentityProvider as an example.
+func TestTLSSpecKubeBuilderValidationSupervisor_Parallel(t *testing.T) {
+	testCases := []struct {
+		name               string
+		customResourceYaml []byte
+		customResourceName string
+		expectedError      string
+	}{
+		{
+			name: "should disallow certificate authority data source with missing name",
+			customResourceYaml: []byte(here.Doc(`
+			---
+			apiVersion: idp.supervisor.pinniped.dev/v1alpha1
+			kind: OIDCIdentityProvider
```

Comment:
These are namespaces resources, so they are typically created in the Supervisor namespace (`env.SupervisorNamespace` in an integration test),

Reply:
This test ensures that the validation in the API spec is working as expected and any future changes don't accidentally violate the current set of assumptions/preconditions. Reconciliation of these resources is not what is being tested. For that reason, it should be OK for these resources to be created in the default namespace. 
I am going to resolve this comment, but please "unresolve" if we need to discuss more or if I may have missed something.